lib/rules/no-restricted-imports.js

```diff
@@ -107,40 +116,66 @@ module.exports = {
             return memo;
         }, {});
 
-        // if no imports are restricted we don"t need to check
-        if (Object.keys(restrictedPaths).length === 0 && restrictedPatterns.length === 0) {
-            return {};
-        }
-
         const restrictedPatternsMatcher = ignore().add(restrictedPatterns);
 
-        /**
-         * Checks to see if "*" is being used to import everything.
-         * @param {Set.<string>} importNames Set of import names that are being imported
-         * @returns {boolean} whether everything is imported or not
-         */
-        function isEverythingImported(importNames) {
-            return importNames.has("*");
-        }
-
         /**
          * Report a restricted path.
+         * @param {string} importSource path of the import
+         * @param {Map<string:Object} importNames Map of import names that are being imported
          * @param {node} node representing the restricted path reference
          * @returns {void}
          * @private
          */
-        function reportPath(node) {
-            const importSource = node.source.value.trim();
-            const customMessage = restrictedPathMessages[importSource] && restrictedPathMessages[importSource].message;
+        function reportPath(importSource, importNames, node) {
```

Comment:
It would be nice to rename `reportPath` to `check...` something, as it's now more than just report.

Reply:
How about `checkRestrictedPathAndReport`?

Reply:
Sounds good! :+1:

Reply:
Done!